### PR TITLE
ssg_test_suite features

### PR DESCRIPTION
**Pull request by maage**
#### Description:

- More debugging, maybe excessively so
- allow to run parallel podman containers
- allow to use other user than root in testing (might require `oscap-ssh` modifications), I guess you normally do not use root user as main ssh target user, you go via other user and then sudo
- some simplifications
- use list to run command
- use shlex to join commands
- use persistent container cache to speed up package installs (has risks)
- change containers to use systemd
- add ssh keys to `admin` user

You just need to ensure automatus jobs do not clash:
- own `--container-ssh-port`
- own `--logdir`
- own `--datastream`
- own `--container`
- and you might need to create cache volume before

#### Rationale:

Modern hw can have multiple CPU and this allows to run multiple podman jobs parallel using automatus slice functionality. Containers can share cache volume and subsequent runs are much faster as there is no need to download packages and related data every time.

Using lists / shlex to manage commands means it is more secure and no need to implement parts.

Adding some more logging helps to diagnose problems easier and understand what is going on. Especially logging Env is very nice, to see what ever is available and what are their values. It could be optimized to log base env w/ macros only once and then per rule additions per test. It seems `Rule` needs to have `__repr__`.

Try to change setup to match actual environments more closely.

#### Review Hints:

This is still WIP I guess. I have no idea how to test this and I don't want to break all the other testing.

At least:
- functionality needs to be added behind proper options in automatus
- I'm not sure about `--debug` implementation, functionality is much needed, but when running parallel, it does not work so easily
- logging should not be excessive
- commit flow is not really flowing

At least at this time there was not anything too great at `codeclimate` issues to specially fix.

### CODE REVIEW

**Comment by maage** [comment]
Had all kinds of buggy sections, I hope less now. Need to use simpler repr. Handled some codeclimate issues, maybe.

**Comment by maage** [comment]
I'm out of good/simple ideas how to handle 3 codeclimate issues left. I can create BaseTestEnv and move simple stuff to there, but does it really help, IDK. And scp one direction and another, IDK, current 4 methods seem like simplest now.

File: tests/ssg_test_suite/common.py [hunk 1]
```diff
@@ -321,6 +310,9 @@ def load_rule_and_env(rule_dir_path, env_yaml, product=None):
     local_env_yaml['rule_title'] = rule.title
     local_env_yaml['products'] = prodtypes
 
+    logging.debug('Rule {}'.format(pprint.pformat(rule)))
+    logging.debug('Env {}'.format(pprint.pformat(local_env_yaml)))
```
Comment (jan-cerny) [inline tests/ssg_test_suite/common.py:314]: It's a large amount of text and there is only little amount of useful data.

**Review by jan-cerny — COMMENTED** [review]
@maage There is a lot of excellent things in this PR that greatly improve the testing tool!

I would prefer if this would be split into multiple smaller PRs. That would make the code easier to review and would allow us to focus more on the changes.

**Comment by mildas** [comment]
I agree with @jan-cerny 
There is a lot of great stuff that I would merge immediately - formatting, code readability, commands as list, or simplifications.
However, then there's the controversial stuff as `admin` as default user. Automatus is now ready only for `root` user. Changing it is going to cause problems. Even if it would be solved for Automatus, then it might be test scenarios' problems.
Splitting it to multiple PRs - debug, container, admin as default, and formatting+simplification - will make review and discussion easier.

**Pull request closed without merging**